File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/models/RelationshipModel.java
package expo.modules.contacts.models;
import android.database.Cursor;
import android.provider.ContactsContract;

import expo.modules.contacts.EXColumns;

import static android.provider.ContactsContract.*;

public class RelationshipModel extends BaseModel {

    @Override
    public String getContentType() {
        return CommonDataKinds.Relation.CONTENT_ITEM_TYPE;
    }

    @Override
    public String getDataAlias() {
        return "name";
    }

    @Override
    protected String getLabelFromCursor(Cursor cursor) {
        String label = super.getLabelFromCursor(cursor);
        if (label != null)
            return label;
        switch (cursor.getInt(cursor.getColumnIndex(EXColumns.TYPE))) {
        case CommonDataKinds.Relation.TYPE_ASSISTANT:
            return "assistant";
        case CommonDataKinds.Relation.TYPE_BROTHER:
            return "bother";
        case CommonDataKinds.Relation.TYPE_CHILD:
            return "child";
        case CommonDataKinds.Relation.TYPE_DOMESTIC_PARTNER:
            return "domesticPartner";
        case CommonDataKinds.Relation.TYPE_FATHER:
            return "father";
        case CommonDataKinds.Relation.TYPE_FRIEND:
            return "friend";
        case CommonDataKinds.Relation.TYPE_MANAGER:
            return "manager";
        case CommonDataKinds.Relation.TYPE_MOTHER:
            return "mother";
        case CommonDataKinds.Relation.TYPE_PARENT:
            return "parent";
        case CommonDataKinds.Relation.TYPE_PARTNER:
            return "partner";
        case CommonDataKinds.Relation.TYPE_REFERRED_BY:
            return "referredBy";
        case CommonDataKinds.Relation.TYPE_RELATIVE:
            return "relative";
        case CommonDataKinds.Relation.TYPE_SISTER:
            return "sister";
        case CommonDataKinds.Relation.TYPE_SPOUSE:
            return "spouse";
        default:
            return "unknown";
        }
    }

}

File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/models/RelationshipModel.kt
package expo.modules.contacts.models

import android.database.Cursor
import android.provider.ContactsContract.CommonDataKinds
import expo.modules.contacts.Columns

class RelationshipModel : BaseModel() {
  override val contentType: String = CommonDataKinds.Relation.CONTENT_ITEM_TYPE
  override val dataAlias: String = "name"

  override fun getLabelFromCursor(cursor: Cursor): String {
    val label = super.getLabelFromCursor(cursor)
    return label
      ?: when (cursor.getInt(cursor.getColumnIndexOrThrow(Columns.TYPE))) {
        CommonDataKinds.Relation.TYPE_ASSISTANT -> "assistant"
        CommonDataKinds.Relation.TYPE_BROTHER -> "bother"
        CommonDataKinds.Relation.TYPE_CHILD -> "child"
        CommonDataKinds.Relation.TYPE_DOMESTIC_PARTNER -> "domesticPartner"
        CommonDataKinds.Relation.TYPE_FATHER -> "father"
        CommonDataKinds.Relation.TYPE_FRIEND -> "friend"
        CommonDataKinds.Relation.TYPE_MANAGER -> "manager"
        CommonDataKinds.Relation.TYPE_MOTHER -> "mother"
        CommonDataKinds.Relation.TYPE_PARENT -> "parent"
        CommonDataKinds.Relation.TYPE_PARTNER -> "partner"
        CommonDataKinds.Relation.TYPE_REFERRED_BY -> "referredBy"
        CommonDataKinds.Relation.TYPE_RELATIVE -> "relative"
        CommonDataKinds.Relation.TYPE_SISTER -> "sister"
        CommonDataKinds.Relation.TYPE_SPOUSE -> "spouse"
        else -> "unknown"
      }
  }
}
